### ISSUE TITLE
Detekt gradle task is not executed upon check

### ISSUE THREAD

**Issue by marschwar** [arturbosch/detekt#1640]
## Expected Behavior
The detekt task should automatically be executed when the user runs `gradle check` or `gradle build`

## Observed Behavior
The detekt task can be run by using `gradle detekt` but it is not triggered on any of the tasks mentioned above.

## Steps to Reproduce
Use any gradle project available and run `gradle clean build`

## Context
Even though there is a test for exactly this action it does not behave the same way when using the plugin. 
Knowing that this works perfectly fine with the ktlint gradle plugin, I did some digging. Since the check dependency in the both plugins are essentially configured the same way, it had to be something else. Turns out that the way the plugin is added to the project makes a difference.

NOT WORKING
```
plugins {
    id "io.gitlab.arturbosch.detekt" version "1.0.0-RC14"
}
```

WORKING
```
buildscript {
    dependencies {
        classpath "io.gitlab.arturbosch.detekt:detekt-gradle-plugin:1.0.0-RC14"
    }
}

allprojects {
  apply plugin: "io.gitlab.arturbosch.detekt"
}
```

If someone could please explain or provide me with pointers on what the difference is, I can try to fix it. The most annoying part is, that it is not reproducible in the tests.

**Comment by 3flex** [comment]
Good pickup, thanks for the detailed report.

I expect the reason is the logic used to create the task dependency at https://github.com/arturbosch/detekt/blob/b1bac0ba85043641cd243197d2a7546b121dd6c3/detekt-gradle-plugin/src/main/kotlin/io/gitlab/arturbosch/detekt/DetektPlugin.kt#L54

If the `check` task is not defined when the `detekt` task is created then adding the dependency won't succeed. Probably the different methods of applying the plugins lead to different outcomes with ordering of task creation or similar.

Not sure yet how to solve but this would hopefully provide a starting point. Any PR has to take task configuration avoidance into account as the detekt task is created lazily.

**Comment by 3flex** [comment]
Didn't mean to step on your toes here (hopefully you hadn't started working on a fix), but I wanted to get this fixed pretty quickly, so I've submitted a PR. Would appreciate if you could try it out? Seemed to fix it for me.

**Comment by marschwar** [comment]
> Didn't mean to step on your toes 
Not at all. I am rather curious what you had to change. Will look at it later tonight.
